actionpack/lib/action_dispatch/middleware/reloader.rb

```diff
@@ -1,54 +1,10 @@
 module ActionDispatch
-  # ActionDispatch::Reloader provides prepare and cleanup callbacks,
-  # intended to assist with code reloading during development.
-  #
-  # Prepare callbacks are run before each request, and cleanup callbacks
-  # after each request. In this respect they are analogs of ActionDispatch::Callback's
-  # before and after callbacks. However, cleanup callbacks are not called until the
-  # request is fully complete -- that is, after #close has been called on
-  # the response body. This is important for streaming responses such as the
-  # following:
-  #
-  #     self.response_body = -> (response, output) do
-  #       # code here which refers to application models
-  #     end
-  #
-  # Cleanup callbacks will not be called until after the response_body lambda
-  # is evaluated, ensuring that it can refer to application models and other
-  # classes before they are unloaded.
+  # ActionDispatch::Reloader wraps the request with the ActiveSupport::Reloader
+  # callbacks, intended to assist with code reloading during development.
   #
   # By default, ActionDispatch::Reloader is included in the middleware stack
   # only in the development environment; specifically, when +config.cache_classes+
-  # is false. Callbacks may be registered even when it is not included in the
-  # middleware stack, but are executed only when <tt>ActionDispatch::Reloader.prepare!</tt>
-  # or <tt>ActionDispatch::Reloader.cleanup!</tt> are called manually.
-  #
+  # is false.
   class Reloader < Executor
```

Comment:
This is an interesting question... if all methods in a class are deprecated, does that mean that the class itself is deprecated? :grimacing: